sherpa-onnx/csrc/offline-tts-zipvoice-frontend.cc

```diff
@@ -1,386 +0,0 @@
-      }
-    }
-  }
-}
-
-static void TokenizeTag(
-    const std::string &words,
-    const std::unordered_map<std::string, int32_t> &token2id,
-    std::vector<int64_t> *tokens_ids, std::vector<std::string> *tokens) {
-  // in zipvoice tags are all in upper case
-  std::string tag = ToUpperAscii(words);
-  if (token2id.count(tag)) {
-    tokens_ids->push_back(token2id.at(tag));
-    tokens->push_back(tag);
-  } else {
-    SHERPA_ONNX_LOGE("Skip unknown tag %s", tag.c_str());
-  }
-}
-
-static void TokenizePinyin(
-    const std::string &words, const cppinyin::PinyinEncoder *pinyin_encoder,
-    const std::unordered_map<std::string, int32_t> &token2id,
-    std::vector<int64_t> *tokens_ids, std::vector<std::string> *tokens) {
-  // words are in the form of <ha3>, <ha4>
-  std::string pinyin = words.substr(1, words.size() - 2);
-  if (!pinyin.empty()) {
-    if (pinyin[pinyin.size() - 1] == '5') {
-      pinyin = pinyin.substr(0, pinyin.size() - 1);  // remove the tone
-    }
-    if (pinyin_encoder->ValidPinyin(pinyin, "number" /*tone*/)) {
-      ProcessPinyin(pinyin, pinyin_encoder, token2id, tokens_ids, tokens);
-    } else {
-      SHERPA_ONNX_LOGE("Invalid pinyin %s", pinyin.c_str());
-    }
-  }
-}
-
-OfflineTtsZipvoiceFrontend::OfflineTtsZipvoiceFrontend(
-    const std::string &tokens, const std::string &data_dir,
-    const std::string &pinyin_dict,
-    const OfflineTtsZipvoiceModelMetaData &meta_data, bool debug /*= false*/)
-    : debug_(debug), meta_data_(meta_data) {
-  std::ifstream is(tokens);
-  token2id_ = ReadTokens(is);
-  if (meta_data_.use_pinyin) {
-    pinyin_encoder_ = std::make_unique<cppinyin::PinyinEncoder>(pinyin_dict);
-  } else {
-    pinyin_encoder_ = nullptr;
-  }
-  if (meta_data_.use_espeak) {
-    // We should copy the directory of espeak-ng-data from the asset to
-    // some internal or external storage and then pass the directory to
-    // data_dir.
-    InitEspeak(data_dir);
-  }
-}
-
-template <typename Manager>
-OfflineTtsZipvoiceFrontend::OfflineTtsZipvoiceFrontend(
-    Manager *mgr, const std::string &tokens, const std::string &data_dir,
-    const std::string &pinyin_dict,
-    const OfflineTtsZipvoiceModelMetaData &meta_data, bool debug)
-    : debug_(debug), meta_data_(meta_data) {
-  auto buf = ReadFile(mgr, tokens);
-  std::istrstream is(buf.data(), buf.size());
-  token2id_ = ReadTokens(is);
-  if (meta_data_.use_pinyin) {
-    auto buf = ReadFile(mgr, pinyin_dict);
-    std::istringstream iss(std::string(buf.begin(), buf.end()));
-    pinyin_encoder_ = std::make_unique<cppinyin::PinyinEncoder>(iss);
-  } else {
-    pinyin_encoder_ = nullptr;
-  }
-  if (meta_data_.use_espeak) {
-    // We should copy the directory of espeak-ng-data from the asset to
-    // some internal or external storage and then pass the directory to
-    // data_dir.
-    InitEspeak(data_dir);
-  }
-}
-
-std::vector<TokenIDs> OfflineTtsZipvoiceFrontend::ConvertTextToTokenIds(
```

Comment:
Have you read all this logic? You just removed all of them? If you don't like cppinyin, you can replace it with jieba + table lookup, but I think you should be careful to remove the whole frontend.

Reply:
OK, [S1], [S2] is for zipvoice dialog, have not supported yet. Well, it's up to you, most of time, it is you who maintain this repo.